modules/api-gateway/iam.tf

```diff
@@ -1,27 +1,14 @@
 data "aws_region" "current" {}
 
-resource "aws_iam_user" "api-gw-user" {
-  count = var.create_user ? 1 : 0
-
-  name = var.iam_username
-}
-
-resource "aws_iam_access_key" "api-gw-ak" {
-  count = var.create_user ? 1 : 0
-
-  user    = aws_iam_user.api-gw-user[0].name
-  pgp_key = var.pgp_key
-}
-
-resource "aws_iam_user_policy" "api-gw-policy" {
-  count = var.create_user ? 1 : 0
-
-  name = var.policy_name
-  user = aws_iam_user.api-gw-user[0].name
+module "api_iam_user" {
+  source = "/Users/Vika/Documents/projects/terraform-aws-modules/modules/aws-iam-user"
 
+  create_user = var.create_iam_user
+  username    = var.iam_username
```

Comment:
lets get rid of iam_username variable and instead generate username based on api-gateway name, something like "{var.name}-api-gtw-usr"